readthedocs/core/signals.py

```diff
@@ -57,8 +58,39 @@ def decide_if_cors(sender, request, **kwargs):  # pylint: disable=unused-argumen
         if domain.exists():
             return True
 
+    # Check for Embed API, allowing CORS on public projects
+    # since they are already public
+    if request.path_info.startswith('/api/v2/embed'):
+        url = request.GET.get('url')
+        if url:
+            unresolved = unresolve(url)
+            project = unresolved.project
+            version_slug = unresolved.version_slug
+        else:
+            project_slug = request.GET.get('project', None)
+            version_slug = request.GET.get('version', None)
+            project = Project.objects.filter(slug=project_slug).first()
+
+        if project and version_slug:
+            # This is from IsAuthorizedToViewVersion,
+            # we should abstract is a bit perhaps?
+            has_access = (
+                Version.objects
+                .public(
+                    user=request.user,
```

Comment:
I don't think this is the filter you are looking for. IIRC, when passing `user=` to the `.public` method it will include private versions as well 😞

Reply:
I found where I mentioned this as well: https://github.com/readthedocs/readthedocs-corporate/pull/1109#pullrequestreview-573442794

Reply:
Yeah, you should just omit the user